src/transport/RendezvousSession.cpp

```diff
@@ -280,7 +325,19 @@ CHIP_ERROR RendezvousSession::HandleSecureMessage(PacketBuffer * msgBuf)
 
     msgBuf->ConsumeHead(headerSize);
 
-    mDelegate->OnRendezvousMessageReceived(msgBuf);
+    if (payloadHeader.GetProtocolID() == Protocols::kChipProtocol_NetworkProvisioning &&
+        payloadHeader.GetMessageType() == NetworkProvisioningMsgTypes::kIPAddressAssigned)
+    {
+        err = HandleIPAddressAssignment(msgBuf);
+        SuccessOrExit(err);
+
+        mDelegate->OnRendezvousStatusUpdate(RendezvousSessionDelegate::NetworkProvisioningSuccess);
```

Comment:
Can we have the OnRendezvousStatusUpdate Sucess or Failure emitted from the adjacent part of the code ?
for example I would do:
```
CHIP_ERROR RendezvousSession::HandleIPAddressAssignment(System::PacketBuffer * buffer)
 {
     if (!IPAddress::FromString(Uint8::to_const_char(buffer->Start()), buffer->DataLength(), mDeviceAddress))
     {
         return CHIP_ERROR_INVALID_ADDRESS;
     }
     return CHIP_NO_ERROR;
 }
```

And in this part of the code I will do something like:

```
err = HandleIPAddressAssignment(msgBuf);
VerifyOrExit(CHIP_NO_ERROR == err), mDelegate->OnRendezvousStatusUpdate(RendezvousSessionDelegate::NetworkProvisioningFailed));

mDelegate->OnRendezvousStatusUpdate(RendezvousSessionDelegate::NetworkProvisioningSuccess);
```

Reply:
Or both in the other functions. I don't really mind as long are adjacent.